runtime/src/loader_utils.rs

```diff
@@ -116,6 +116,7 @@ pub fn load_upgradeable_buffer<T: Client>(
     let program = load_program_from_file(name);
     let buffer_pubkey = buffer_keypair.pubkey();
     let buffer_authority_pubkey = buffer_authority_keypair.pubkey();
+    let program_buffer_bytes = UpgradeableLoaderState::size_of_programdata(program.len());
```

Comment:
Why is this using `size_of_programdata`? In the instruction creator, it's allocating enough for `size_of_buffer`: https://github.com/solana-labs/solana/blob/613f0ff2260b117a32406489e2b2ab535e6997e1/sdk/program/src/bpf_loader_upgradeable.rs#L135

So unless I'm missing something, this should be `UpgradeableLoaderState::size_of_buffer(program.len())`

Reply:
yes. Should be size_of_buffer. I will create a PR to fix this. Thanks!

Reply:
In fact, if we look at the log more carefully 

> In this example, the program data is 102496 bytes, while the account's data is
> 102533 bytes. 

We should use buffer_metadata, 37 bytes. not program_metadata 45bytes.